src/vm/argdestination.h

```diff
@@ -42,31 +45,17 @@ class ArgDestination
         return dac_cast<PTR_VOID>(dac_cast<TADDR>(m_base) + m_offset);
     }
 
-#if defined(UNIX_AMD64_ABI) && defined(FEATURE_UNIX_AMD64_STRUCT_PASSING)
+#if (defined(UNIX_AMD64_ABI) && defined(FEATURE_UNIX_AMD64_STRUCT_PASSING)) || defined(_TARGET_ARM64_)
 
     // Returns true if the ArgDestination represents a struct passed in registers.
     bool IsStructPassedInRegs()
     {
         LIMITED_METHOD_CONTRACT;
+#if defined(_TARGET_ARM64_)
+        return m_argLocDescForStructInRegs != NULL;
```

Comment:
That's not the case. It is set only for structs in registers, for others, it is NULL. The ArgDestination::m_argLocDescForStructInRegs is set in the constructor and we pass ArgIterator::GetArgLocDescForStructInRegs result to the constructor. And the ArgIterator::GetArgLocDescForStructInRegs looks like this:
```C++
    ArgLocDesc* GetArgLocDescForStructInRegs()
    {
#if defined(UNIX_AMD64_ABI) && defined(FEATURE_UNIX_AMD64_STRUCT_PASSING)
        return m_hasArgLocDescForStructInRegs ? &m_argLocDescForStructInRegs : NULL;
#else
        return NULL;
#endif
    }
```
So it only returns non-NULL if the argument is a struct passed in registers.